missalemeum/kalendar/rules.py

```diff
@@ -138,6 +138,11 @@ def rule_shift_conflicting_1st_class_feasts(
     # If there are two feasts with 1st class, the one with lower priority on Precedence Table is shifted to the first
     # day where there is no 1st and 2nd class feast.
 
+    # The feast of the Immaculate Conception of the Blessed Virgin Mary, however,
+    # is preferred to the Sunday of Advent on which it may occur. (General Rubrics, 15)
+    if date_ == date(date_.year, 12, 8) and date_.weekday() == SUNDAY:
+        return [match(observances, PATTERN_SANCTI)], [match(observances, PATTERN_TEMPORA)], []
```

Comment:
Can you add a test for this?

Reply:
There are already some tests covering the cases like the Sunday on December 8, Sunday on another day, December 8 not on Sunday.

`test_given_date_contains_proper_day_ids`:
```
((1907, 12, 8), [c.TEMPORA_ADV2_0], [c.SANCTI_12_08], [c.TEMPORA_ADV2_0]),
((1912, 12, 8), [c.TEMPORA_ADV2_0], [c.SANCTI_12_08], [c.TEMPORA_ADV2_0]),
((1913, 12, 8), [c.TEMPORA_ADV2_1], [c.SANCTI_12_08], []),
```

`test_conflicts`:
```
((2019, 12, 8), [c.SANCTI_12_08], [c.TEMPORA_ADV2_0]),
((2019, 9, 8), [c.TEMPORA_PENT13_0], [c.SANCTI_09_08]),
```
(the second one is a Sunday not on December 8, which proves that the new code didn't force a commemoration of the Sunday)

`test_calculated_commemorations`:
```
# In Conceptione Immaculata Beatæ Mariæ Virginis, commemoration of the Sunday
((2019, 12, 8), …
```

Is something missing? By the way this is how I discovered the problem after fixing the original issue. These tests were failing.

Reply:
Fair enough. Thank you!